### PR TITLE
- Fixed determining package status of directory to go through the UTI tr...

### PR DESCRIPTION
**Pull request by MrNoodle**
...ee.

The old code was a bit simplistic and results in certain folders being treated as packages when they shouldn't. This new method actually goes through the UTI tree looking for the package type which should be far more accurate. Note, though, that getting the UTI declaration is a bit expensive so I added a static cache. If you want, you can stick this as an ivar though this info should be the same system-wide.

### CODE REVIEW

**Comment by MrNoodle** [comment]
BTW, do I need to be concerned about that method being called on multiple threads? If so, I should add some locking of the UTI cache.

**Comment by mikeabdullah** [comment]
OK, here's the bit I don't get. Why can't we just use `UTTypeConformsTo()`?

**Comment by MrNoodle** [comment]
Err, because I'm an idiot? For some reason when I initially tested it it seemed to only checked if it directly conformed but my tests now indicate that it seems to work all the way up the hierarchy. So, I guess my initial test was bad. I'll fix it and re-push.

**Comment by MrNoodle** [comment]
Should I rebase these into a single commit?

**Comment by mikeabdullah** [comment]
If you prefer to, go for it. I'm perfectly happy with having the full commit history (don't know why some people insists on ditching it!)

File: ConnectionKit/CK2FileManager.m [hunk 1]
```diff
@@ -330,39 +330,27 @@ - (BOOL)ck2_getResourceValue:(out id *)value forKey:(NSString *)key error:(out N
...
-                    else if (status != noErr)
-                    {
-                        NSLog(@"Error getting app info for extension for URL %@: %s", [self absoluteString], GetMacOSStatusCommentString(status));
-                    }
-                    else
+                    if (UTTypeConformsTo((CFStringRef)uti, CFSTR("com.apple.package")))
                     {
                         *value = @YES;
-                        return YES;
+                        break;
```
Comment (mikeabdullah) [inline ConnectionKit/CK2FileManager.m:347]: So one concern, am I right in thinking this `break` will cause `baseUTIs` to be leaked?

Comment (mikeabdullah) [inline ConnectionKit/CK2FileManager.m:347]: Actually shut up Mike, the `for` loop is in the right place to take care of that.

**Comment by mikeabdullah** [comment]
I'm happy, so merge whenever you are :)

**Comment by MrNoodle** [comment]
I'll leave the commits as is then. Go ahead and merge. Or should I do it?

**Pull request merged**